### ISSUE TITLE
L027 Expects table alias for current_timestamp

### ISSUE THREAD

**Issue by dmateusp** [alanmcruickshank/sqlfluff#418]
```sql
select
    ta.column_a,
    current_timestamp as column_b,
    tb.column_c
from table_a as ta
join table_b as tb
    on ta.id = tb.id
```

gives 

```
L:   3 | P:   5 | L027 | Unqualified reference 'current_timestamp' found in
                       | select with more than one referenced table/view.
```

I don't think the parser sees the current_timestamp as a function, which is a problem here
```
naked_identifier_ReSegment:         [28](1, 3, 5)       'current_timestamp'
```

For example in the case of:
```sql
day(ta.column_d) as day_d
```

the parsed segment is
```
FunctionSegment:
```

and that does not return a lint error, which is correct

**Comment by alanmcruickshank** [comment]
Good catch.

At the moment there's not exception for "bare" functions like `current_timestamp` which aren't followed by brackets.

We should probably add another "set" in the parser to allow people to configure these. @dmateusp can you think of any other functions which also don't have brackets after them?

**Comment by dmateusp** [comment]
I looked for a few more, found this Oracle list of ANSI functions: https://docs.oracle.com/cd/E57185_01/IRWUG/ch12s04s05.html, they have a table with a column which is for "which parameters the functions take" so it's easy to find nullary functions.

We could start with `row_number`, `current_timestamp`, `current_time`, `current_date`. 

Interestingly the Snowflake USAGE NOTES for `current_timestamp` actually specify that it can be called without brackets to comply with ANSI https://docs.snowflake.com/en/sql-reference/functions/current_timestamp.html#usage-notes

**Comment by dmateusp** [comment]
Thought I'd point out as well that Snowflake as some nullary functions like `current_role` which cannot be called without brackets

So we really just need to make an exception for these functions that can be called without brackets
